### ISSUE TITLE
The instructions for patching Configure could use a bit of an update

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9945]
Migrated from [rt.perl.org#70203](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=70203) (status was 'resolved')

Searchable as RT70203$

**Comment by p5pRT** [comment]
### From @obra

#### Created by @obra



We should clarify that any change \_committed\_ to Configure needs
to be extracted into metaconfig\. also that metaconfig lives in git\, not sourceforge\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=low

Site configuration information for perl 5.10.0:

Configured by Debian Project at Thu Oct  1 22:38:45 UTC 2009.

Summary of my perl5 (revision 5 version 10 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.6.24-23-server, archname=i486-linux-gnu-thread-multi
    uname='linux vernadsky 2.6.24-23-server #1 smp wed apr 1 22:22:14 utc 2009 i686 gnulinux '
    config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=i486-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.10 -Darchlib=/usr/lib/perl/5.10 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.10.0 -Dsitearch=/usr/local/lib/perl/5.10.0 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Ud_ualarm -Uusesfio -Uusenm -DDEBUGGING=-g -Doptimize=-O2 -Duseshrplib -Dlibperl=libperl.so.5.10.0 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion='', gccversion='4.4.1', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /usr/lib64
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/libc-2.10.1.so, so=so, useshrplib=true, libperl=libperl.so.5.10.0
    gnulibc_version='2.10.1'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -g -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.10.0:
    /home/jesse/git/bps/prophet/lib
    /home/jesse/git/bps/sd/lib
    /etc/perl
    /usr/local/lib/perl/5.10.0
    /usr/local/share/perl/5.10.0
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.10
    /usr/share/perl/5.10
    /usr/local/lib/site_perl
    .


Environment for perl 5.10.0:
    HOME=/home/jesse
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/jesse/git/bps-private/git-tools/:/home/jesse/git/bps/sd/bin/:/home/jesse/bin/:/usr/local/bin:/opt/local/bin:/opt/local/sbin:/opt/ec2/bin:/home/jesse/android-sdk-linux_x86-1.5_r3/tools:/home/jesse/git/github/mine/bin:/home/jesse/git/other/git-wt/bin/:/home/jesse/git/bps-private/git-tools/:~/git/bps/sd/bin/:~/bin/:/usr/local/bin:/opt/local/bin:/opt/local/sbin:/opt/ec2/bin:~/android-sdk-linux_x86-1.5_r3/tools:~/git/github/mine/bin:~/git/other/git-wt/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games
    PERL5LIB=/home/jesse/git/bps/prophet/lib:/home/jesse/git/bps/sd/lib
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @tux

On Tue\, 03 Nov 2009 13&#8203;:08&#8203;:09 \-0800\, Jesse \(via RT\)
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> We should clarify that any change \_committed\_ to Configure needs
> to be extracted into metaconfig\. also that metaconfig lives in git\, not sourceforge\.

In defense \.\.\.

http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/
http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/blob/HEAD&#8203;:/README is already
up\-to\-date and mentions

$ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig

\-\- 
H\.Merijn Brand  http&#8203;://tux\.nl      Perl Monger  http&#8203;://amsterdam\.pm\.org/
using & porting perl 5\.6\.2\, 5\.8\.x\, 5\.10\.x\, 5\.11\.x on HP\-UX 10\.20\, 11\.00\,
11\.11\, 11\.23\, and 11\.31\, OpenSuSE 10\.3\, 11\.0\, and 11\.1\, AIX 5\.2 and 5\.3\.
http&#8203;://mirrors\.develooper\.com/hpux/           http&#8203;://www\.test\-smoke\.org/
http&#8203;://qa\.perl\.org      http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @obra




On Tue  3\.Nov'09 at 23&#8203;:41&#8203;:59 \-0800\, H\. Merijn Brand via RT wrote&#8203;:
> On Tue\, 03 Nov 2009 13&#8203;:08&#8203;:09 \-0800\, Jesse \(via RT\)
> \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> 
> > We should clarify that any change \_committed\_ to Configure needs
> > to be extracted into metaconfig\. also that metaconfig lives in git\, not sourceforge\.
>
> In defense \.\.\.
> 
> http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/
> http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/blob/HEAD&#8203;:/README is already
> up\-to\-date and mentions
> 
> $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig

Apologies\! My statement was about&#8203;:

\# Note&#8203;: this Configure script was generated automatically\. Rather than
\# working with this copy of Configure\, you may wish to get metaconfig\.
\# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
\#     svn co https://svn.sourceforge.net/svnroot/dist/trunk/dist
\#
\#
\# Though this script was generated by metaconfig\, it is OK to send
\# patches against it\. It's up to the Configure pumpkin to backport
\# the patch to the metaunits if it is accepted\.
\# See Porting/pumpkin\.pod for more information on metaconfig\.

in blead's "Configure"

**Comment by p5pRT** [comment]
### From @tux

On Wed\, 4 Nov 2009 16&#8203;:29&#8203;:08 \-0500\, Jesse Vincent \<jesse@&#8203;fsck\.com> wrote&#8203;:

> 
> 
> 
> On Tue  3\.Nov'09 at 23&#8203;:41&#8203;:59 \-0800\, H\. Merijn Brand via RT wrote&#8203;:
> > On Tue\, 03 Nov 2009 13&#8203;:08&#8203;:09 \-0800\, Jesse \(via RT\)
> > \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> > 
> > > We should clarify that any change \_committed\_ to Configure needs
> > > to be extracted into metaconfig\. also that metaconfig lives in git\, not sourceforge\.
> >
> > In defense \.\.\.
> > 
> > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/
> > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/blob/HEAD&#8203;:/README is already
> > up\-to\-date and mentions
> > 
> > $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig
> 
> Apologies\! My statement was about&#8203;:
> 
> \# Note&#8203;: this Configure script was generated automatically\. Rather than
> \# working with this copy of Configure\, you may wish to get metaconfig\.
> \# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
> \#     svn co https://svn.sourceforge.net/svnroot/dist/trunk/dist
> \#
> \#
> \# Though this script was generated by metaconfig\, it is OK to send
> \# patches against it\. It's up to the Configure pumpkin to backport
> \# the patch to the metaunits if it is accepted\.
> \# See Porting/pumpkin\.pod for more information on metaconfig\.
> 
> in blead's "Configure"

Ah\, correct\. I will address that shortly\!

\-\- 
H\.Merijn Brand  http&#8203;://tux\.nl      Perl Monger  http&#8203;://amsterdam\.pm\.org/
using & porting perl 5\.6\.2\, 5\.8\.x\, 5\.10\.x\, 5\.11\.x on HP\-UX 10\.20\, 11\.00\,
11\.11\, 11\.23\, and 11\.31\, OpenSuSE 10\.3\, 11\.0\, and 11\.1\, AIX 5\.2 and 5\.3\.
http&#8203;://mirrors\.develooper\.com/hpux/           http&#8203;://www\.test\-smoke\.org/
http&#8203;://qa\.perl\.org      http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
### From @jkeenan

On Wed Nov 04 23&#8203;:51&#8203;:57 2009\, hmbrand wrote&#8203;:
> On Wed\, 4 Nov 2009 16&#8203;:29&#8203;:08 \-0500\, Jesse Vincent \<jesse@&#8203;fsck\.com>
> wrote&#8203;:
> 
> >
> >
> >
> > On Tue  3\.Nov'09 at 23&#8203;:41&#8203;:59 \-0800\, H\. Merijn Brand via RT wrote&#8203;:
> > > On Tue\, 03 Nov 2009 13&#8203;:08&#8203;:09 \-0800\, Jesse \(via RT\)
> > > \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> > >
> > > > We should clarify that any change \_committed\_ to Configure needs
> > > > to be extracted into metaconfig\. also that metaconfig lives in
> git\, not sourceforge\.
> > >
> > > In defense \.\.\.
> > >
> > > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/
> > > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/blob/HEAD&#8203;:/README is
> already
> > > up\-to\-date and mentions
> > >
> > > $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig
> >
> > Apologies\! My statement was about&#8203;:
> >
> > \# Note&#8203;: this Configure script was generated automatically\. Rather
> than
> > \# working with this copy of Configure\, you may wish to get
> metaconfig\.
> > \# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
> > \#     svn co https://svn.sourceforge.net/svnroot/dist/trunk/dist
> > \#
> > \#
> > \# Though this script was generated by metaconfig\, it is OK to send
> > \# patches against it\. It's up to the Configure pumpkin to backport
> > \# the patch to the metaunits if it is accepted\.
> > \# See Porting/pumpkin\.pod for more information on metaconfig\.
> >
> > in blead's "Configure"
> 
> Ah\, correct\. I will address that shortly\!
> 

Has this question about how/where we say changes to Configure need to be
committed been addressed?

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @tux

On Sat\, 03 Aug 2013 10&#8203;:07&#8203;:04 \-0700\, "James E Keenan via RT"
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> On Wed Nov 04 23&#8203;:51&#8203;:57 2009\, hmbrand wrote&#8203;:
> > On Wed\, 4 Nov 2009 16&#8203;:29&#8203;:08 \-0500\, Jesse Vincent \<jesse@&#8203;fsck\.com>
> > wrote&#8203;:
> > 
> > >
> > >
> > >
> > > On Tue  3\.Nov'09 at 23&#8203;:41&#8203;:59 \-0800\, H\. Merijn Brand via RT wrote&#8203;:
> > > > On Tue\, 03 Nov 2009 13&#8203;:08&#8203;:09 \-0800\, Jesse \(via RT\)
> > > > \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> > > >
> > > > > We should clarify that any change \_committed\_ to Configure needs
> > > > > to be extracted into metaconfig\. also that metaconfig lives in
> > git\, not sourceforge\.
> > > >
> > > > In defense \.\.\.
> > > >
> > > > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/
> > > > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/blob/HEAD&#8203;:/README is
> > already
> > > > up\-to\-date and mentions
> > > >
> > > > $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig
> > >
> > > Apologies\! My statement was about&#8203;:
> > >
> > > \# Note&#8203;: this Configure script was generated automatically\. Rather
> > than
> > > \# working with this copy of Configure\, you may wish to get
> > metaconfig\.
> > > \# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
> > > \#     svn co https://svn.sourceforge.net/svnroot/dist/trunk/dist
> > > \#
> > > \#
> > > \# Though this script was generated by metaconfig\, it is OK to send
> > > \# patches against it\. It's up to the Configure pumpkin to backport
> > > \# the patch to the metaunits if it is accepted\.
> > > \# See Porting/pumpkin\.pod for more information on metaconfig\.
> > >
> > > in blead's "Configure"
> > 
> > Ah\, correct\. I will address that shortly\!
> > 
> 
> Has this question about how/where we say changes to Configure need to be
> committed been addressed?

I changed that to
\-\-8\<\-\-\-
\# Note&#8203;: this Configure script was generated automatically\. Rather than
\# working with this copy of Configure\, you may wish to get metaconfig\.
\# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
\#     svn co https://svn.code.sf.net/p/dist/code/trunk/dist
\#
\# Though this script was generated by metaconfig from metaunits\, it is
\# OK to send patches against Configure itself\. It's up to the Configure
\# pumpkin to backport the patch to the metaunits if it is accepted\.
\# For more information on patching Configure\, see pod/perlhack\.pod
\#
\# The metaunits are also available from the public git repository&#8203;:
\#     http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/ or
\#     $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig
\#
\# See Porting/pumpkin\.pod for more information on metaconfig\.
\-\->8\-\-\-

back in

commit 7f5516d11013641bb723aba27b6cb26c845b096c
Author&#8203;: H\.Merijn Brand \<merijn@&#8203;nb09\.procura\.nl>
Date&#8203;:   Thu Nov 5 15&#8203;:32&#8203;:11 2009 \+0100

&nbsp;    Mention git repo in Configure's header


I think that the \*middle\* section is enough\, right?

> Thank you very much\.
> Jim Keenan

\-\- 
H\.Merijn Brand  http&#8203;://tux\.nl   Perl Monger  http&#8203;://amsterdam\.pm\.org/
using perl5\.00307 \.\. 5\.19   porting perl5 on HP\-UX\, AIX\, and openSUSE
http&#8203;://mirrors\.develooper\.com/hpux/        http&#8203;://www\.test\-smoke\.org/
http&#8203;://qa\.perl\.org   http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
### From @jkeenan

On Sat Sep 07 09&#8203;:18&#8203;:41 2013\, hmbrand wrote&#8203;:
> On Sat\, 03 Aug 2013 10&#8203;:07&#8203;:04 \-0700\, "James E Keenan via RT"
> \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> 
> > On Wed Nov 04 23&#8203;:51&#8203;:57 2009\, hmbrand wrote&#8203;:
> > > On Wed\, 4 Nov 2009 16&#8203;:29&#8203;:08 \-0500\, Jesse Vincent \<jesse@&#8203;fsck\.com>
> > > wrote&#8203;:
> > > 
> > > >
> > > >
> > > >
> > > > On Tue  3\.Nov'09 at 23&#8203;:41&#8203;:59 \-0800\, H\. Merijn Brand via RT wrote&#8203;:
> > > > > On Tue\, 03 Nov 2009 13&#8203;:08&#8203;:09 \-0800\, Jesse \(via RT\)
> > > > > \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> > > > >
> > > > > > We should clarify that any change \_committed\_ to Configure needs
> > > > > > to be extracted into metaconfig\. also that metaconfig lives in
> > > git\, not sourceforge\.
> > > > >
> > > > > In defense \.\.\.
> > > > >
> > > > > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/
> > > > > http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/blob/HEAD&#8203;:/README is
> > > already
> > > > > up\-to\-date and mentions
> > > > >
> > > > > $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig
> > > >
> > > > Apologies\! My statement was about&#8203;:
> > > >
> > > > \# Note&#8203;: this Configure script was generated automatically\. Rather
> > > than
> > > > \# working with this copy of Configure\, you may wish to get
> > > metaconfig\.
> > > > \# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
> > > > \#     svn co https://svn.sourceforge.net/svnroot/dist/trunk/dist
> > > > \#
> > > > \#
> > > > \# Though this script was generated by metaconfig\, it is OK to send
> > > > \# patches against it\. It's up to the Configure pumpkin to backport
> > > > \# the patch to the metaunits if it is accepted\.
> > > > \# See Porting/pumpkin\.pod for more information on metaconfig\.
> > > >
> > > > in blead's "Configure"
> > > 
> > > Ah\, correct\. I will address that shortly\!
> > > 
> > 
> > Has this question about how/where we say changes to Configure need to be
> > committed been addressed?
> 
> I changed that to
> \-\-8\<\-\-\-
> \# Note&#8203;: this Configure script was generated automatically\. Rather than
> \# working with this copy of Configure\, you may wish to get metaconfig\.
> \# The dist package \(which contains metaconfig\) is available via SVN&#8203;:
> \#     svn co https://svn.code.sf.net/p/dist/code/trunk/dist
> \#
> \# Though this script was generated by metaconfig from metaunits\, it is
> \# OK to send patches against Configure itself\. It's up to the Configure
> \# pumpkin to backport the patch to the metaunits if it is accepted\.
> \# For more information on patching Configure\, see pod/perlhack\.pod
> \#
> \# The metaunits are also available from the public git repository&#8203;:
> \#     http&#8203;://perl5\.git\.perl\.org/metaconfig\.git/ or
> \#     $ git clone git&#8203;://perl5\.git\.perl\.org/metaconfig\.git metaconfig
> \#
> \# See Porting/pumpkin\.pod for more information on metaconfig\.
> \-\->8\-\-\-
> 
> back in
> 
> commit 7f5516d11013641bb723aba27b6cb26c845b096c
> Author&#8203;: H\.Merijn Brand \<merijn@&#8203;nb09\.procura\.nl>
> Date&#8203;:   Thu Nov 5 15&#8203;:32&#8203;:11 2009 \+0100
> 
>     Mention git repo in Configure's header
> 
> 
> I think that the \*middle\* section is enough\, right?
> 

Yes\.  Thanks for the response\.  Marking ticket resolved\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
